src/java.base/share/classes/java/util/FormatItem.java

```diff
@@ -143,9 +143,9 @@ static final class FormatItemDecimal implements FormatConcatItem {
             this.zeroDigit = dfs.getZeroDigit();
             this.minusSign = dfs.getMinusSign();
             this.digitOffset = this.zeroDigit - '0';
-            int length = DecimalDigits.INSTANCE.size(value);
+            int length = DecimalDigits.stringSize(value);
             this.digits = new byte[length];
-            DecimalDigits.INSTANCE.digits(value, this.digits, length, PUT_CHAR_DIGIT);
+            DecimalDigits.getCharsLatin1(value, length, this.digits);
```

Comment:
Sorry missed this one in last round of review. This is wrong if you mix number fornat items and chinese segments in a String Template.

Reply:
(Just getting back from vacation) That's why we have PUT_CHAR_DIGIT. I don't really like this refactoring because of this. I get it but it needs to take UTF16 into account.

Reply:
Try again (been on vacation). Not sure I agree with this refactoring. You need to account for UTF16, hence the PUT_CHAR_DIGIT. Recommend you revert and add getCharsLatin1 using the original implementation.

Reply:
Have a great vacation! Enjoy your time off! I made a lot of changes, including removing the use of MethodHandler in FormatItem. I'm not in a hurry. You can handle the review after you come back from vacation.